src/nix/path-info.cc

```diff
@@ -152,11 +155,15 @@ struct CmdPathInfo : StorePathsCommand, MixJSON
             pathLen = std::max(pathLen, store->printStorePath(storePath).size());
 
         if (json) {
-            std::cout << pathInfoToJSON(
+            auto pair = pathInfoToJSON(
                 *store,
                 // FIXME: preserve order?
                 StorePathSet(storePaths.begin(), storePaths.end()),
-                showClosureSize).dump();
+                showClosureSize);
+            auto json = pair.first;
+            auto resultCode = pair.second;
+            std::cout << json.dump();
+            throw Exit(resultCode);
```

Comment:
I find this pattern odd but it seems how it's done.
Would be nice to set a variable on the Command class itself that is ultimately returned.
Could be a follow-up which is broader.

Reply:
Haskell and iirc python do it this way. It's a valid solution, but I could get behind a solution without exceptions.
That way an application level concern (exit codes) doesn't get mixed with the same mechanism for domain-level error handling.
Let's stick with this for now.